[mbr] Add Apple sample

lambdageek opened the pull request:
Hot reload for iOS Simulator and Mac Catalyst

Contributes to #44806

lambdageek committed:
[mbr] Start adding an Apple sample

Runs, but doesn't do deltas yet

lambdageek committed:
[mbr] Add an apple sample

Works on iOS simulator and Mac Catalyst

CoffeeFlux commented on src/mono/sample/mbr/DeltaHelper/DeltaHelper.cs:
```diff
@@ -48,7 +48,8 @@ public class DeltaHelper {
         {
-            UpdateMethod (assm, dmeta_data, dil_data, dpdb_data);
+            // UpdateMethod (assm, dmeta_data, dil_data, dpdb_data);
+	    System.Reflection.Metadata.AssemblyExtensions.ApplyUpdate (assm, dmeta_data, dil_data, dpdb_data);
```
Indentation looks wrong here, are you mixing tabs and spaces?

lambdageek replied:
bah. yes. and also I meant to delete a bunch of the old code in here.

CoffeeFlux reviewed (commented):
At a glance, you appear to have indentation issues in most of these files. I pointed out one such example.

lambdageek committed:
delete unused DeltaHelper code

Now that System.Reflection.Metadata.AssemblyExtensions.ApplyUpdate is in the
BCL officially, I don't need the reflection-based invoke code anymore

lambdageek committed:
fix indentation

CoffeeFlux reviewed (approved):
Didn't test locally, but generally LGTM.
